json/src/main/scala/fs2/data/json/tokens.scala

```diff
@@ -55,7 +62,298 @@ object Token {
     def jsonRepr: String = value
   }
   case class StringValue(value: String) extends Token("string") {
-    def jsonRepr: String = s""""$value""""
+    def jsonRepr: String = {
+      val rendered = new StringBuilder
+      rendered.append('"')
+      renderString(value, 0, rendered)
+      rendered.append('"')
+      rendered.result()
+    }
+  }
+
+  private final val hex = "0123456789abcdef"
+
+  @tailrec
+  def renderString(s: String, idx: Int, builder: StringBuilder): Unit =
+    if (idx < s.length) {
+      val nextEscape = s.indexWhere(c => c > 127 || Character.isISOControl(c) || "\\/\b\f\n\r\t\"".contains(c), idx)
+      if (nextEscape >= 0) {
+        if (nextEscape > 0) {
+          builder.append(s.substring(idx, nextEscape))
+        }
+        val c = s(nextEscape)
+        (c: @switch) match {
+          case '\\' =>
+            builder.append("\\\\")
+          case '/' =>
+            builder.append("\\/")
+          case '\b' =>
+            builder.append("\\b")
+          case '\f' =>
+            builder.append("\\f")
+          case '\n' =>
+            builder.append("\\n")
+          case '\r' =>
+            builder.append("\\r")
+          case '\t' =>
+            builder.append("\\t")
+          case '"' =>
+            builder.append("\\\"")
+          case _ =>
+            // escape non ascii or control characters
+            builder
+              .append("\\u")
+              .append(hex((c >> 12) & 0x0f))
+              .append(hex((c >> 8) & 0x0f))
+              .append(hex((c >> 4) & 0x0f))
+              .append(hex(c & 0x0f))
+        }
+        renderString(s, nextEscape + 1, builder)
+      } else {
+        // append the rest of the string and we are done
+        builder.append(s.substring(idx))
+      }
+    }
+
+  private val nullValue =
+    Stream.emit(DocEvent.Text("null"))
+
+  private val trueValue =
+    Stream.emit(DocEvent.Text("true"))
+
+  private val falseValue =
+    Stream.emit(DocEvent.Text("false"))
+
+  private final val FirstObjectKey = 0
+  private final val ObjectKey = 1
+  private final val ObjectValue = 2
+  private final val FirstArrayValue = 3
+  private final val ArrayValue = 4
+
+  implicit object renderable extends Renderable[Token] {
```

Comment:
Any specific reason to make the "pretty" renderer the implicit default and not compact? I would have expected it rather the other way round…

Reply:
It was driven by the compact having a custom pipe, so I thought that when people use the `prettyPrint` pipe, they would actually expect pretty printing being the default.